.github/workflows/release.yml

```diff
@@ -0,0 +1,57 @@
+name: Release
+
+# Runs manually
+on:
+  workflow_dispatch:
+      
+jobs:
+  release:
+    name: Release on Sonatype OSS
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Set up Apache Maven Central
+        uses: actions/setup-java@v3
+        with:
+          java-version: 11
+          distribution: 'temurin'
+          cache: 'maven'
+          server-id: ossrh # Value of the distributionManagement/repository/id field of the pom.xml
+          server-username: SONATYPE_USERNAME # env variable for username in deploy
+          server-password: SONATYPE_PASSWORD # env variable for password in deploy
+          gpg-private-key: ${{ secrets.GPG_PRIVATE_KEY }} # Value of the GPG private key to import
+          gpg-passphrase: GPG_PASSPHRASE # env variable for GPG private key passphrase
+
+      - name: Publish to Apache Maven Central
+        run: |
+          mvn -DskipTests deploy -Dgpg.passphrase=${{ secrets.GPG_PASSPHRASE }}
```

Comment:
Also add `--settings .mvn/settings.xml` to use our settings from git and you can remove the server settungs from `java-setup`.

Reply:
Why are we skipping the tests? We should never have a release when tests are not successful.

Reply:
I forgot to mention that some tests were failing because they are from the other projects e.g Ide-Settings and only available on the remote repository and since I used my own groupId to test the release process those tests were not found because I didn't deploy the other projects. I tried to deploy Ide-Settings as I did with maven-parent and IDE but I didn't find any pom.xml in it so I have no Idea how to deploy it to my remote repository?